File: Python/04- POO/03- Atividades Pontuada/01- Atividade Pontuada/models/funcionario.py
from abc import ABC, abstractmethod
from models.fisica import Fisica
from models.enum.setor import Setor
from models.enum.sexo import Sexo
from models.endereco import Endereco


class Funcionario(Fisica, ABC):
    def __init__(self, nome, telefone, email, endereco: Endereco, cpf, rg, data_nascimento, sexo, matricula: str, setor: Setor, salario: float) -> None:
        super().__init__(nome, telefone, email, endereco, cpf, rg, data_nascimento, sexo)
        self.matricula = matricula
        self.setor = setor
        self.salario = salario

    @abstractmethod
    def salario_final(self) -> float:
        pass

    def __str__(self) -> str:
        return (f"{super().__str__()}"
               f"\nMatrícula: {self.matricula}"
               f"\nSetor: {self.setor.value}"
               f"\nSalário: {self.salario}"
               f"\nSalário Final: {self.salario_final()}"
               )

File: Java/02- Relacionando Classes/17- Produto/Funcionario.java

package com.mycompany.mavenproject7;

public abstract class Funcionario extends Fisica{
    
    protected String cpf;
    protected String rg;
    protected String matricula;
    protected Setor setor;
    protected double salario;

    public Funcionario(String cpf, String rg, String matricula, Setor setor, double salario, Sexo sexo, EstadoCivil estadocivil, String dataNascimento, String nome, String telefone, String email, Endereco endereco) {
        super(sexo, estadocivil, dataNascimento, nome, telefone, email, endereco);
        this.cpf = cpf;
        this.rg = rg;
        this.matricula = matricula;
        this.setor = setor;
        this.salario = salario;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public String getRg() {
        return rg;
    }

    public void setRg(String rg) {
        this.rg = rg;
    }

    public String getMatricula() {
        return matricula;
    }

    public void setMatricula(String matricula) {
        this.matricula = matricula;
    }

    public Setor getSetor() {
        return setor;
    }

    public void setSetor(Setor setor) {
        this.setor = setor;
    }

    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }

    @Override
    public String toString() {
        return super.toString() +
                "\n=====Funcionário===== " + 
                "\nCPF: " + cpf + 
                "\nRG: " + rg + 
                "\nMatricula: " + matricula + 
                "\nSetor: " + setor.getTexto() + 
                "\nSalário: " + salario;
    }
    
}